ote/src/clj/ote/services/transport.clj

```diff
@@ -99,31 +101,19 @@
            (delete! db ::t-service/transport-service {::t-service/id id}))
          (http/transit-response id)))))
 
-
-(defn- ensure-transport-operator-for-group [db {:keys [title id] :as ckan-group}]
-  ;; FIXME: this should be middleware, not relying on client to make a POST request
-  (tx/with-transaction db
-    (let [operator (get-transport-operator db {::t-operator/ckan-group-id id})]
-      (or operator
-          (and id
-          ;; FIXME: what if name changed in CKAN, we should update?
-          (insert! db ::t-operator/transport-operator
-                   {::t-operator/name title
-                    ::t-operator/ckan-group-id id}))))))
-
-
 (defn get-user-transport-operators-with-services [db groups user]
-  (let [operators (map #(ensure-transport-operator-for-group db %) groups)
+  (let [operators (map #(get-transport-operator db {::t-operator/ckan-group-id (:id %)}) groups)
         operator-ids (into #{} (map ::t-operator/id) operators)
         operator-services (get-transport-services db {::t-service/transport-operator-id (op/in operator-ids)})]
-    (map (fn [{id ::t-operator/id :as operator}]
-            {:transport-operator operator
+          (map (fn [{id ::t-operator/id :as operator}]
+           (when id
```

Comment:
Can `id` be `nil`? Why is the `when` check here... is it ok that the output list contains nil values?
If not, use `keep`

Reply:
nope, its not ok. I'll take a look